File: 3958-minimum-removals-to-balance-array/minimum-removals-to-balance-array.java
class Solution
{
    public int minRemoval(int[] nums, int k) 
    {
        int count=Integer.MAX_VALUE;
        Arrays.sort(nums);

        for(int i=0;i<nums.length;i++)
        {
            long val = (long)nums[i]*k;

            int lb=i;
            int ub=nums.length-1;
            while(lb<=ub)
            {
                int mid = (lb+ub)/2;
                if(val>=nums[mid])
                    lb=mid+1;
                else
                    ub=mid-1;
            }

            count = Math.min(count,i+nums.length-lb);
        }

        return count;
    }
}

File: 3958-minimum-removals-to-balance-array/minimum-removals-to-balance-array.cpp
class Solution {
public:
    int minRemoval(vector<int>& nums, int k) 
    {
        sort(nums.begin(),nums.end());
        int n=nums.size();
        int l=0;
        int maxx=1;
        for(int r=0;r<n;r++)
        {
            while((long long)nums[r]>(long long)nums[l]*k)
            {
                l++;
            }

            maxx=max(maxx,r-l+1);
        }

        return n-maxx;
    }
};